api/v1/atlasnetworkpeering_types_test.go

```diff
@@ -0,0 +1,19 @@
+package v1 // nolint: dupl
+
+import (
+	"testing"
+)
+
+func TestNetworkPeeringProjectRefCELValidations(t *testing.T) {
+	launchProjectRefCELTests(
```

Comment:
q: what is being asserted here? `launchProjectRefCELTests` has no indication what test cases are being executed.

Reply:
This is interesting, because all cases were always the same for all the users. The thing is, the only change was the dual reference was used by one or another underlying CRD type, but that does not change the behaviour or correctness of how the dual ref is handled.


We could fix this in a separate PR, I think we can have a single test for this in a single place and just keep on adding underlaying CRDs to test as carriers of the dual reference to be tested.
WDYT?